changelog/_unreleased/2025-04-04-fix-product-media-import-with-whitespace-in-url.md

```diff
@@ -0,0 +1,16 @@
+---
+title: fix product media import with whitespace in url
+issue: 5868
+author: p.dinkhoff
+author_email: p.dinkhoff@shopware.com
+author_github: @p.dinkhoff
+---
+# Core
+* Changed: URLs generated by the `MediaUrlGenerator` now encode whitespaces as `%20` in version `v6.8.0.0`. This ensures valid URLs in media and Import/Export operations.
```

Comment:
```suggestion
* Changed: URLs generated by the `MediaUrlGenerator` will properly encode the file path to produce valid URLs in version `v6.8.0.0` (e.g. replace whitespaces as `%20`). The current implementation relies on the caller to do proper URL encoding with the returned string.
```

Reply:
Regarding:
> This ensures valid URLs in media and Import/Export operations.
What about implementing a workaround for that, like all the other places in 6.7 we could manually encode the URLs for now by calling the proper method / service.

Reply:
I personally would not enter the way it was before that. That can be confusing, while reading the changelog.